include/boost/json/detail/basic_parser.hpp

```diff
@@ -189,16 +189,15 @@ class basic_parser
     number num_;
     error_code ec_;
     detail::stack st_;
-    std::size_t max_depth_ = 32;
-    // how many levels deeper the parser can go
-    std::size_t depth_ = max_depth_;
     detail::utf8_sequence seq_;
     unsigned u1_;
     unsigned u2_;
     bool more_; // false for final buffer
     bool complete_ = false; // true on complete parse
     const char* end_;
     parse_options opt_;
+    // how many levels deeper the parser can go
+    std::size_t depth_ = opt_.max_depth;
```

Comment:
could call it `depth_remain_`. or not.

Reply:
Meh, not too important. If you want to rename it, go ahead.

Reply:
I don't feel strongly about it